lib/app_profiler.rb

```diff
@@ -60,17 +68,77 @@ module Viewer
   mattr_reader :after_process_queue, default: nil
 
   class << self
-    def run(*args, &block)
-      Profiler.run(*args, &block)
+    def run(*args, backend: nil, **kwargs, &block)
+      orig_backend = self.backend
+      begin
+        self.backend = backend if backend
+        profiler.run(*args, **kwargs, &block)
+      rescue BackendError
+        yield
```

Comment:
The premise behind that is that we don't want profiling to cause a request that would otherwise succeed to fail.

Maybe to avoid obliquely swallowing the error, we can just log that this occurred here.

Reply:
Hm, I wonder what happens when you supply bad values to things like incorrect modes today. Do we need to rescue other error types?